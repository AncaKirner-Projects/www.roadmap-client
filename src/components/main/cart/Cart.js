import React, { Component } from 'react';
import Typography from '@material-ui/core/Typography';
import { connect } from 'react-redux';
import styled from 'styled-components';
import CartElement from './CartElement';

const totalPrice = (products) => {
  let total = 0;
  products.forEach(product => { total += product.price * product.quantity; });
  return total;
};

const TotalStyle = styled(Typography)`
  padding: 40px 9% 0px 0px;
`;

class Cart extends Component {
  state = {}

  render() {
    const { products } = this.props;

    return (
      <React.Fragment>
        <Typography variant="h5" component="h3">
          My Cart
        </Typography>
        <div>
          {
            products && products.map(product => <CartElement key={product.id} product={product} />)
          }
        </div>
        <TotalStyle variant="h5" align="right">
          Total: &nbsp;
          {totalPrice(products)}
          &nbsp; RON
        </TotalStyle>
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => ({
  products: state.cart.products
});

export default connect(mapStateToProps)(Cart);
