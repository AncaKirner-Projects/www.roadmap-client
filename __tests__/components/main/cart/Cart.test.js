import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import Cart from '../../../../src/components/main/cart/Cart';
// import ActionTypes from '../../../src/store/actions/actionTypes';

const middlewares = [thunk];
const mockStore = configureMockStore();

describe('<Cart />', () => {
  const store = mockStore({
    cart: {
      products: [
        { id: 1, prod_name: 'Prod1' }
      ]
    }
  });

  it('renders an element', () => {
    const wrapper = shallow(<Cart store={store} />);
    const component = wrapper.dive();

    expect(toJson(component)).toMatchSnapshot();
  });
});
