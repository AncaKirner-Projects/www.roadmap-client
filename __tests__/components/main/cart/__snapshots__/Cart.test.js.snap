// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> renders an element 1`] = `
<Fragment>
  <WithStyles(Typography)
    component="h3"
    variant="h5"
  >
    My Cart
  </WithStyles(Typography)>
  <div>
    <Connect(CartElement)
      key="1"
      product={
        Object {
          "id": 1,
          "prod_name": "Prod1",
        }
      }
    />
  </div>
  <Styled(WithStyles(Typography))
    align="right"
    variant="h5"
  >
    Total:  
    NaN
      RON
  </Styled(WithStyles(Typography))>
</Fragment>
`;
